src/CachePlugin.php

```diff
@@ -183,11 +236,12 @@ protected function isCacheable(ResponseInterface $response)
         if (!in_array($response->getStatusCode(), [200, 203, 300, 301, 302, 404, 410])) {
             return false;
         }
-        if (!$this->config['respect_cache_headers']) {
-            return true;
-        }
-        if ($this->getCacheControlDirective($response, 'no-store') || $this->getCacheControlDirective($response, 'private')) {
-            return false;
+
+        $cacheableDirectives = $this->extractDirectives($this->config['respect_response_cache_directives'], $this->noCacheFlags);
```

Comment:
$cacheableDirectives can only be one of these four (no matter what). Is this really the intention?

```php
 ['no-cache', 'private']
 ['private']
 ['no-cache']
[]
```